runhouse/resources/hardware/cluster.py

```diff
@@ -798,15 +819,15 @@ def _run_cli_commands_on_cluster_helper(self, commands: List[str]):
     def restart_server(
         self,
         _rh_install_url: str = None,
-        resync_rh: bool = True,
+        resync_rh: Optional[bool] = None,
         restart_ray: bool = True,
         restart_proxy: bool = False,
         logs_level: str = None,
     ):
         """Restart the RPC server.
 
         Args:
-            resync_rh (bool): Whether to resync runhouse. (Default: ``True``)
+            resync_rh (bool): Whether to resync runhouse. Specifying False will not sync Runhouse under any circumstance. If it is None, then it will sync if Runhouse is not installed on the cluster or if locally it is installed as editable. (Default: ``None``)
```

Comment:
I think we don't need to specify the behavior for False or True, should be pretty self explanatory.

```suggestion
            resync_rh (bool): Whether to resync runhouse. By default, it will sync if Runhouse is not installed on the cluster or if locally it is installed as editable. (Default: ``None``)
```